### PR TITLE
Add rebalance quadratic equation apis

### PR DESCRIPTION
**Pull request by halink0803**

### CODE REVIEW

**Comment by halink0803** [comment]
[Implement rebalance_quadratic APIs](https://trello.com/c/FgoRMvtb/133-implement-rebalancequadratic-apis)

File: data/storage/bolt.go [hunk 1]
```diff
@@ -1681,3 +1692,92 @@ func (self *BoltStorage) GetPWIEquationV2() (metric.PWIEquationRequestV2, error)
...
+
+//StorePendingRebalanceQuadratic store pending data (stand for rebalance quadratic equation) to db
+//data byte for json {"KNC": {"a": 0.9, "b": 1.2, "c": 1.4}}
+func (self *BoltStorage) StorePendingRebalanceQuadratic(data []byte) error {
+	timepoint := common.GetTimepoint()
+	err := self.db.Update(func(tx *bolt.Tx) error {
+		b := tx.Bucket([]byte(PENDING_REBALANCE_QUADRATIC))
+		c := b.Cursor()
+		_, v := c.First()
+		if v != nil {
```
Comment (favadi) [inline data/storage/bolt.go]: checking the key should be better.

File: data/storage/bolt.go [hunk 2]
```diff
@@ -1681,3 +1692,92 @@ func (self *BoltStorage) GetPWIEquationV2() (metric.PWIEquationRequestV2, error)
...
+}
+
+//GetPendingRebalanceQuadratic return pending rebalance quadratic equation
+func (self *BoltStorage) GetPendingRebalanceQuadratic() (metric.RebalanceQuadraticRequest, error) {
+	var result metric.RebalanceQuadraticRequest
+	err := self.db.View(func(tx *bolt.Tx) error {
+		b := tx.Bucket([]byte(PENDING_REBALANCE_QUADRATIC))
+		c := b.Cursor()
+		_, v := c.First()
+		if v == nil {
```
Comment (favadi) [inline data/storage/bolt.go]: checking the key should be better.

File: data/storage/bolt.go [hunk 3]
```diff
@@ -1681,3 +1692,92 @@ func (self *BoltStorage) GetPWIEquationV2() (metric.PWIEquationRequestV2, error)
...
+}
+
+//GetRebalanceQuadratic return current confirm rebalance quadratic equation
+func (self *BoltStorage) GetRebalanceQuadratic() (metric.RebalanceQuadraticRequest, error) {
+	var result metric.RebalanceQuadraticRequest
+	err := self.db.View(func(tx *bolt.Tx) error {
+		b := tx.Bucket([]byte(REBALANCE_QUADRATIC))
+		c := b.Cursor()
+		_, v := c.Last()
+		return json.Unmarshal(v, &result)
```
Comment (favadi) [inline data/storage/bolt.go]: Should we check if v != nil?

File: http/rebalance_quadratic.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+//RejectRebalanceQuadratic reject pending configuration for rebalance quadratic function
+func (h *HTTPServer) RejectRebalanceQuadratic(c *gin.Context) {
+	_, ok := h.Authenticated(c, []string{}, []Permission{ConfirmConfPermission})
+	if !ok {
+		return
+	}
+	err := h.metric.RemovePendingRebalanceQuadratic()
+	if err != nil {
```
Comment (favadi) [inline http/rebalance_quadratic.go]: please use if err := xxx; err != nil form.

File: http/rebalance_quadratic_test.go [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+				}
+			}	
+		}`
+	)
+
+	common.RegisterInternalActiveToken(common.Token{ID: "KNC"})
+	common.RegisterInternalActiveToken(common.Token{ID: "ETH"})
+
+	tmpDir, err := ioutil.TempDir("", "test_rebalance_quadratic")
+
```
Comment (favadi) [inline http/rebalance_quadratic_test.go]: remove this empty line.

**Pull request merged**